### ISSUE TITLE
Overload order matters when implementing SupportsRound

### ISSUE THREAD

**Issue by karlb** [python/mypy#9628]
Using mypy 0.790 on

```python
from fractions import Fraction

round(Fraction(3, 2))
```

yields the result
```
foo.py:3: error: No overload variant of "round" matches argument type "Fraction"
foo.py:3: note: Possible overload variants:
foo.py:3: note:     def round(number: float) -> int
foo.py:3: note:     def [_T] round(number: SupportsRound[_T]) -> int
foo.py:3: note:     <4 more non-matching overloads not shown>
```

The code works and mypy 0.782 did not complain, so this looks like a bug to me.

**Comment by hauntsaninja** [comment]
The stubs for fractions were previously very incomplete, which is why this surfaces now.

It looks like this is fixable in typeshed by just switching the overload definitions around... https://github.com/python/typeshed/pull/4697
As far as I can tell, there isn't a good reason for why this should be necessary, so leaving this open as a bug on mypy

**Comment by hauntsaninja** [comment]
Thinking about it more, I guess overload order does matter even for cases where you'd expect it shouldn't matter because of the spectre of multiple inheritance? (Although NoneType should be final, so maybe in this case order really doesn't matter)

Either way, your original complaint is fixed by the typeshed change :-)

**Comment by karlb** [comment]
Thanks for the quick fix! I assume we can close the issue, then.
